agent/xds/rbac.go

```diff
@@ -576,7 +635,7 @@ func makeRBACRules(
 	for i, rbacIxn := range rbacIxns {
 		var infos []*JWTInfo
 		if isHTTP {
-			infos = collectJWTInfos(rbacIxn)
+			infos = rbacIxn.jwtInfos
```

Comment:
Correct me if I'm wrong, but I don't see a need to carry `rbacIxn.jwtInfos` around to make the Principals in `addJWTPrincipals`.

When I see `rbacIxn.ComputedPrincipal` I would expect JWT-related principals to be already computed and part of that field (similar to how perms are written in `rbacIxn.ComputedPermission`).

Reply:
> The main reason is because this is the ideal place to check whether or not it is isHTTP. After we compute the principals, we have a few places below where we optimize them too.

We actually check `isHTTP` earlier in `intentionToIntermediateRBACForm`. `rixn.jwtInfos` is only set when `isHTTP == true`.

I think all the computation for JWT principals should be happening in `(*rbacIntention) FlattenPrincipal` similar to the changes you've already made for `(*rbacPermission) Flatten()`. `rbacIxn` already contains all the information required to generate JWT principals (and it is safe for L4 since it can be a no-op if `len(rixn.jwtInfos) == 0`) .

> Ideally we want to addJWTPrincipals when everything else is already done rather than interfere with existing stuff

I disagree with this because `FlattenPrincipal` is where all logic for principal injection seems to be happening. JWT should be part of it and not a special case.